### ISSUE TITLE
Problems with deferred works if other server has wrong settings, error log in HTTP request

### ISSUE THREAD

**Issue by IronBug** [friendica/friendica#8653]
Greetings.
I use the latest developer Friendica (Friendica 'Red Hot Poker' 2020.06-dev - 1348) version. I was debugging a strange problem: I had all works as deferred for one server, despite the messages were delivered.
It seems, the problem is not on my side. I get strange http answer from the other server (also Friendica). here's the debug of message I get in answer to mail sending (I masked the real address):

> 2020-05-17T14:02:09Z worker [DEBUG]: https://social.userserver.net/inbox/username: 200 HTTP/2 200
server: nginx
date: Sun, 17 May 2020 14:00:05 GMT
content-type: text/html; charset=UTF-8
vary: Accept-Encoding
x-powered-by: PHP/7.4.5
x-account-management-status: none
>
><pre><b>replaceMacros</b>: unable to write file /home/www/friendica/view/smarty3/compiled/wrt5ec14365ac89a7_28134661</pre> {"worker_id":"837c5b9","worker_cmd":"APDelivery"} - {"file":"CurlResult.php","line":133,"function":"__construct", "uid":"458ecf","process_id":25258}

The mail is sent successfully, but the strange thing is that I see the other server error message (possibly write fail to smarty3/compiled) in the HTTP answer to my server! 

And this causes the process of mail delivery to fail and as a consequence I get deferred work with empty task:

> 2020-05-17T14:02:09Z worker [INFO]: No delivery data for  mail - Item ID: 17 - Contact ID: 4908 {"worker_id":"837c5b9","worker_cmd":"Delivery"} - {"file":"Delivery.php","line":181,"function":"execute","uid":"458ecf","process_id":25258}
This is the second part of the quest. And this leads to long lasting retrials of delivery of empty task. 

It's not a showstopper but the problem with error log in HTTP is a serious bug.

**Comment by MrPetovan** [comment]
Thanks for the report. I’m not sure what would be different for you if the error message was removed from the HTTP response?

**Comment by annando** [comment]
Yeah, it's slidly weird that this error message is part of the content. And I think that we should have a look at it, since it reveals the folder structure of the remote side.

**Comment by MrPetovan** [comment]
The error definitely should be muted, but I don't understand what it would change regarding deferred tasks and zombie processes.

**Comment by annando** [comment]
Since error messages are appearing in the header, I guess that the communication came to a halt, since this will most likely invalidate the data.

**Comment by IronBug** [comment]
yes, the problem with exposing the structure should be fixed for security reasons.

but I'm more concerned with deferred zombie-tasks. the other server admin removed problem with access to smarty. and now I get proper HTTP responses. but zombie-tasks remained.
I cannot guess out how they appear. and why I have this problem with only one server (that is accessible). 
and I found a record 'Contact 4908 is marked for archival' after some period of zombie-tasks. I don't want it to become archival, how can I dismiss 'archiving'?
I attach an excerpt from logs that shows one mail sending trace (the task and the process that performs the task). maybe you could help me to figure out what's wrong.
[friendica-zombie-task.log](https://github.com/friendica/friendica/files/4641092/friendica-zombie-task.log)

**Comment by IronBug** [comment]
maybe the problem of error log in HTTP and zombie-tasks are independent, non coherent bugs, I cannot say for sure. just I hunted down the zombie-tasks problem and discovered the log piece in HTTP.

**Comment by annando** [comment]
I don't know what you mean with "Zombie-Task".

**Comment by IronBug** [comment]
zombie-task (see the top post) is a everlasting deferred task that hangs in deferred tasks until the number of retrials's exceeded. but the problem is that the message is already delivered! the user on other side can see it, no problem. and my server tries to 'deliver' it again and again. and all messages to one certain server become zombies. 

ah, I must add: I found out that on the other side there's also the same trunk version of Friendica. so we're talking about the development versions.

**Comment by IronBug** [comment]
exact versions from the other side:
friendica - 62f3c1f4b8b63bba6117ac50a49387fafd71a90a
addons - 5688629a4b0f70ad80844215f2d2d072f5d210ef
on my side:
friendica - f58d42e317cbf323fb2c802a0cb71a4799863008
addons - 59cfb05c3a53825efe4e4a5c828d12d505f591b5

**Comment by MrPetovan** [comment]
Template engine errors are now been handled in the Friendica log and won't affect displayed output anymore. A side-effect is that the 200 responses you got will become 500 errors.
